networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/connection/ConnectionManagerTest.java

```diff
@@ -156,8 +176,48 @@ public void shouldReconnectWhenPersistentPeerDisconnects() {
     peer.disconnectImmediately(Optional.empty(), true);
 
     assertThat(asyncRunner.hasDelayedActions()).isTrue();
-    asyncRunner.executeQueuedActions();
+
+    // time hasn't advanced enough to trigger a retry
+    asyncRunner.executeDueActions();
+    verify(network, times(1)).connect(PEER1);
+
+    // advance time enough to trigger a retry
+    timeProvider.advanceTimeBySeconds(9);
+    asyncRunner.executeDueActions();
+
     verify(network, times(2)).connect(PEER1);
+
+    // fast disconnection, last disconnection delay should not reset
+    assertThat(manager.getLastStaticPeerReconnectionDelay(PEER1)).isEqualTo(9);
```

Comment:
what is the difference between this test and `shouldRetryConnectionToStaticPeerAfterRetryAndDisconnect`?